Add rebranding changes and few more fixes for encountered issues due to Yocto5 update

hashraf91 opened the pull request:
* setup-flex: remove Sokol keyword from print message
* conf-notes.txt: print only officially supported image
* distro: update variables with new distro name
* The DataSmart.finalize() method has been removed upstream: https://github.com/openembedded/bitbake/commit/584989ed2b5af4e8799571dece0cf94f995ef14e. This was an empty method and now it is completely removed.
* drop ';' delimiter from ROOTFS/IMAGE*COMMAND variables. The variables handling changed upstream: https://github.com/openembedded/openembedded-core/commit/c3365dfd9ddd7fbe70b62e0f11166e57a8ca6f73
* sokol-flex.conf: do not create SPDX by-default

hashraf91 committed:
sdk_codebench_metadata.bbclass: fix failure in get_cb_options

The DataSmart.finalize() method has been removed upstream:
https://github.com/openembedded/bitbake/commit/584989ed2b5af4e8799571dece0cf94f995ef14e

This was an empty method and now it is completely removed. This
fixes the failure in do_populate_sdk.

JIRA-ID: SB-23685

Signed-off-by: Haseeb Ashraf <haseeb_ashraf@mentor.com>

hashraf91 committed:
drop ';' delimiter from SDK/ROOTFS/IMAGE*COMMAND variables

The variables handling changed upstream:
https://github.com/openembedded/openembedded-core/commit/c3365dfd9ddd7fbe70b62e0f11166e57a8ca6f73

JIRA-ID: SB-23685

Signed-off-by: Haseeb Ashraf <haseeb_ashraf@mentor.com>

hashraf91 committed:
setup-flex: remove Sokol keyword from print message

The new name is "Flex OS" only hence removed "Sokol" keyword.

JIRA-ID: SB-23685

Signed-off-by: Haseeb Ashraf <haseeb_ashraf@mentor.com>

hashraf91 committed:
conf-notes.txt: print only officially supported image

production-image is not officially supported image anymore. Hence,
removing it from the notes.

JIRA-ID: SB-23685

Signed-off-by: Haseeb Ashraf <haseeb_ashraf@mentor.com>

hashraf91 committed:
distro: update variables with new distro name

We have not yet updated the DISTRO variable but only the printing
information related to DISTRO. This is an intermediate change which
should be removed once DISTRO is updated with new name.

JIRA-ID: SB-23685

Signed-off-by: Haseeb Ashraf <haseeb_ashraf@mentor.com>

hashraf91 committed:
sokol-flex.conf: do not create SPDX by-default

This has been enabled by-default after Yocto5 update:
https://github.com/openembedded/bitbake/commit/b34032ec22921d4fd0982c17e63732cdbbc1885a

We do not need to create SPDX by-default to save some build time,
as this creates SPDX for each recipe. If required, this can be
enabled back by adding following line to local.conf:
```
INHERIT += "create-spdx"
```

JIRA-ID: SB-23685

Signed-off-by: Haseeb Ashraf <haseeb_ashraf@mentor.com>

Noor-Ahsan commented on meta-sokol-flex-support/recipes-core/meta/archive-release.bb:
```diff
@@ -19,11 +19,13 @@ BINARY_ARTIFACTS_COMPRESSION ?= ""
 ARCHIVE_RELEASE_VERSION ?= "${DISTRO_VERSION}"
-MANIFEST_NAME ?= "${DISTRO}-${ARCHIVE_RELEASE_VERSION}-${MACHINE}"
-EXTRA_MANIFEST_NAME ?= "${DISTRO}-${ARCHIVE_RELEASE_VERSION}"
+MANIFEST_NAME ?= "flex-os-${ARCHIVE_RELEASE_VERSION}-${MACHINE}"
```
we have changed the distro name. see https://github.com/MentorEmbedded/meta-sokol-flex/blob/scarthgap/meta-sokol-flex-distro/conf/distro/sokol-flex.conf#L6. Should we use Distri Name here or Distro and rename Distro as well. lets have a discussion with Awais on this on renaming DISTRO

Noor-Ahsan commented:
Let's discuss Distro name thing with Awais and finalize it based on that discussion

hashraf91 committed:
distro: local.conf.sample: drop splash by-default

We are not using splash screen for our BSPs, hence removing it
from local.conf.sample.

JIRA-ID: SB-23692

Signed-off-by: Haseeb Ashraf <haseeb_ashraf@mentor.com>

ahsanhussain commented:
LGTM apart from the intermediary distro name related stuff. Let's finalize the distro name and remove those commits.

hashraf91 commented:
> LGTM apart from the intermediary distro name related stuff. Let's finalize the distro name and remove those commits.

Squashed the commits for intermediate branding change as discussed. The complete re-branding change will be done in separate PR.
